src/Umbraco.Web.UI.Client/src/views/propertyeditors/blockgrid/prevalue/umbBlockGridAreaEditor.component.js

```diff
@@ -45,46 +75,78 @@
+            initializeSortable();
+            vm.loading = false;
 
-            assetsService.loadJs('lib/sortablejs/Sortable.min.js', $scope).then(onLoaded);
         };
 
-        function onLoaded() {
-            vm.loading = false;
-            initializeSortable();
-        }
-
         function initializeSortable() {
 
-            function _sync(evt) {
-
-                const oldIndex = evt.oldIndex,
-                      newIndex = evt.newIndex;
+            vm.sorterOptions = {
+                resolveVerticalDirection: resolveVerticalDirection,
+                compareElementToModel: (el, modelEntry) => modelEntry.key === el.dataset.areaKey,
+                querySelectModelToElement: (container, modelEntry) => container.querySelector(`[data-area-key='${modelEntry.key}']`),
+                itemHasNestedContainersResolver: () => false,// We never have nested in this case.
+                containerSelector: ".umb-block-grid-area-editor__grid-wrapper",
+                itemSelector: ".umb-block-grid-area-editor__area",
+                placeholderClass: "umb-block-grid-area-editor__area-placeholder",
+                items: vm.model,
+                onSync: onSortSync
+            }
 
-                vm.model.splice(newIndex, 0, vm.model.splice(oldIndex, 1)[0]);
-                
+            function onSortSync() {
+                $scope.$evalAsync();
+                setDirty();
             }
 
-            const gridContainerEl = $element[0].querySelector('.umb-block-grid-area-editor__grid-wrapper');
-
-            const sortable = Sortable.create(gridContainerEl, {
-                sort: true,  // sorting inside list
-                animation: 150,  // ms, animation speed moving items when sorting, `0` — without animation
-                easing: "cubic-bezier(1, 0, 0, 1)", // Easing for animation. Defaults to null. See https://easings.net/ for examples.
-                cancel: '',
-                draggable: ".umb-block-grid-area-editor__area",  // Specifies which items inside the element should be draggable
-                ghostClass: "umb-block-grid-area-editor__area-placeholder",
-                onAdd: function (evt) {
-                    _sync(evt);
-                    $scope.$evalAsync();
-                },
-                onUpdate: function (evt) {
-                    _sync(evt);
-                    $scope.$evalAsync();
-                }
-            });
+            function resolveVerticalDirection(data) {
+
+                /** We need some data about the grid to figure out if there is room to be placed next to the found element */
+                const approvedContainerComputedStyles = getComputedStyle(data.containerElement);
+                const gridColumnGap = Number(approvedContainerComputedStyles.columnGap.split("px")[0]) || 0;
+                const gridColumnNumber = vm.rootLayoutColumns;
+    
+                const foundElColumns = parseInt(data.relatedElement.dataset.colSpan, 10);
```

Comment:
There's a potential for returning a `NaN` here. Consider if we should check for that (I don't know how volatile this particular data-attribute is)

Reply:
Yeah. Well, the col-span is required for these items. It must be there for things to work correctly. So I think its good that we get an error in that case.